dts/bindings/ieee802154/atmel,rf2xx.yaml

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2019, Gerson Fernando Budke
+# SPDX-License-Identifier: Apache-2.0
+
+title: ATMEL AT86RF2xx 802.15.4 Wireless Transceiver
+
+description: |
+    This is a representation of the ATMEL AT86RF2xx wireless transceiver.
+
+compatible: "atmel,rf2xx"
+
+include: spi-device.yaml
+
+properties:
+    hwcaps-mask:
+      type: int
+      required: true
+      description: Hardware capabilities(1 - 2.4G, 2 - Sub.Giga, 3 - Both)
```

Comment:
Yes, you can have it; yes there isn't meaning on do that at first impression; and can be confusing. I agree with your comment about hard-coded too.

This is a provocative field for sure. As you know there are things that can be configured by menuconfig (build time) and others by DT (run-time). This way DT can be used to avoid inflate a 2.4G version (not implemented) and helps to remember that system need to have Sub-Giga. The menuconfig configuration still necessary and DT must not change that. It is important mention that AT86RF215 have both 2.4/Sub-Giga on same package. Yes, 2 HW ifaces on the same SPI, this is future development.

At end, this field can be used to reinforce that drivers matchs with run-time requirements. It can be droped If this isn't necessary that.

Reply:
Ok, then this will need a bit of rework. Basically, we will need to expose some 15.4 macros to DT, a bit like include/dt-bindings/gpio/gpio.h for instance, instead of creating custom caps per-yaml.

It think it's too early at this stage of your driver to expose this. Just get it running already for 2.4Ghz and let's merge it. In the mean time let's fix the macro exposure to DT. At this point you'll be able to improve your driver and expose such field for 2.4/sub-g.